src/main/java/com/cloudant/client/api/CloudantClient.java

```diff
@@ -530,17 +553,31 @@ Response put(URI uri, Object object, boolean newEntity, int writeQuorum, Gson gs
      */
     private void doInit(String scheme, String hostname, int port,
                         String loginUsername, String password, ConnectOptions connectOptions,
-                        String authCookie) {
+                        String authCookie, CookieInterceptor interceptor) {

```

Comment:
Take this `CookieInterceptor` out of the constructor for now and create it from the credentials in this method. Obviously we'll need to expose setters for the interceptors later.